Sources/SwiftProtobuf/BinaryDecoder.swift

```diff
@@ -18,1477 +18,1505 @@
-            value.reserveCapacity(value.count + n / MemoryLayout<Int32>.size)
-            var decoder = BinaryDecoder(forReadingFrom: p, count: n, parent: self)
+      }
+
+      internal mutating func decodeSingularSFixed32Field(value: inout Int32?)
+            throws
+      {
+            guard fieldWireFormat == WireFormat.fixed32 else { return }
             var i: Int32 = 0
-            while !decoder.complete {
-                try decoder.decodeFourByteNumber(value: &i)
-                value.append(Int32(littleEndian: i))
-            }
+            try decodeFourByteNumber(value: &i)
+            value = Int32(littleEndian: i)
             consumed = true
-        default:
-            return
-        }
-    }
-
-    internal mutating func decodeSingularSFixed64Field(value: inout Int64) throws {
-        guard fieldWireFormat == WireFormat.fixed64 else {
-            return
-        }
-        var i: Int64 = 0
-        try decodeEightByteNumber(value: &i)
-        value = Int64(littleEndian: i)
-        consumed = true
-    }
-
-    internal mutating func decodeSingularSFixed64Field(value: inout Int64?) throws {
-        guard fieldWireFormat == WireFormat.fixed64 else {
-            return
-        }
-        var i: Int64 = 0
-        try decodeEightByteNumber(value: &i)
-        value = Int64(littleEndian: i)
-        consumed = true
-    }
-
-    internal mutating func decodeRepeatedSFixed64Field(value: inout [Int64]) throws {
-        switch fieldWireFormat {
-        case WireFormat.fixed64:
+      }
+
+      internal mutating func decodeRepeatedSFixed32Field(value: inout [Int32])
+            throws
+      {
+            switch fieldWireFormat {
+            case WireFormat.fixed32:
+                  var i: Int32 = 0
+                  try decodeFourByteNumber(value: &i)
+                  value.append(Int32(littleEndian: i))
+                  consumed = true
+            case WireFormat.lengthDelimited:
+                  var n: Int = 0
+                  let p = try getFieldBodyBytes(count: &n)
+                  value.reserveCapacity(
+                        value.count + n / MemoryLayout<Int32>.size)
+                  var decoder = BinaryDecoder(
+                        forReadingFrom: p, count: n, parent: self)
+                  var i: Int32 = 0
+                  while !decoder.complete {
+                        try decoder.decodeFourByteNumber(value: &i)
+                        value.append(Int32(littleEndian: i))
+                  }
+                  consumed = true
+            default: return
+            }
+      }
+
+      internal mutating func decodeSingularSFixed64Field(value: inout Int64)
+            throws
+      {
+            guard fieldWireFormat == WireFormat.fixed64 else { return }
             var i: Int64 = 0
             try decodeEightByteNumber(value: &i)
-            value.append(Int64(littleEndian: i))
+            value = Int64(littleEndian: i)
```

Comment:
I hope the formatter didn't make this change!

Reply:
😄 it's the diffs not really lining up, it is different functions with similar bodies confusing things.